pytorch_lightning/callbacks/prediction_writer.py

```diff
@@ -109,7 +109,7 @@ def on_predict_batch_end(
         if not self.interval.on_batch:
             return
         is_distributed = trainer.accelerator_connector.is_distributed
-        batch_indices = trainer.predict_loop.batch_indices if is_distributed else None
+        batch_indices = trainer.predict_loop.prediction_loop.current_batch_indices if is_distributed else None
```

Comment:
`predict_loop.prediction_loop`

is confusing. Can we change the variable names?

`predict_loop.batch_loop`?

Reply:
there is no batch loop for prediction (currently). remember: x_loop means it loops over one instance of x, e.g. epoch_loop runs over a single epoch. 

`predict_loop.epoch_loop?`

Reply:
sure

Reply:
renamed epoch loop
also: #8077